internal/consumer/consumer.go

```diff
@@ -560,6 +588,29 @@ func (i *InventoryConsumer) RebalanceCallback(consumer *kafka.Consumer, event ka
 		i.Logger.Warnf("consumer rebalance event type: %d new partition(s) assigned: %v\n",
 			len(ev.Partitions), ev.Partitions)
 
+		if len(ev.Partitions) > 0 {
+			p := ev.Partitions[0] // there should only be one partition
+			i.lockId = fmt.Sprintf("%s/%d", i.Config.ConsumerGroupID, p.Partition)
+			i.Logger.Infof("Attempting to acquire lock for lockId: %s", i.lockId)
+
+			lockToken, err := i.Retry(func() (string, error) {
```

Comment:
`i.Retry` currently logs failures using the `MsgProcessFailures` metric, which would be misaligned here. We might need to consider updating the function to support a custom metric name to prevent pollution.